pkg/infra/usagestats/service/usage_stats.go

```diff
@@ -66,47 +73,75 @@ func (uss *UsageStats) gatherMetrics(ctx context.Context, metrics map[string]int
 	metrics["stats.usagestats.debug.collect.error.count"] = errC
 }
 
+func (uss *UsageStats) runMetricsFunc(ctx context.Context, fn usagestats.MetricsFunc) (map[string]interface{}, error) {
+	start := time.Now()
+	ctx, span := uss.tracer.Start(ctx, "UsageStats.Gather")
+	fnName := runtime.FuncForPC(reflect.ValueOf(fn).Pointer()).Name()
+	span.SetAttributes("usageStats.function", fnName, attribute.Key("usageStats.function").String(fnName))
+	defer span.End()
+
+	fnMetrics, err := fn(ctx)
+	if err != nil {
+		uss.log.FromContext(ctx).Error("Failed to fetch usage stats from provider", "error", err, "duration", time.Since(start), "function", fnName)
+		span.SetStatus(codes.Error, fmt.Sprintf("failed to fetch usage stats from provider: %v", err))
+		return nil, err
+	}
+
+	uss.log.FromContext(ctx).Debug("Successfully fetched usage stats from provider", "duration", time.Since(start), "function", fnName)
+	return fnMetrics, nil
+}
+
 func (uss *UsageStats) RegisterMetricsFunc(fn usagestats.MetricsFunc) {
 	uss.externalMetrics = append(uss.externalMetrics, fn)
 }
 
-func (uss *UsageStats) sendUsageStats(ctx context.Context) error {
+func (uss *UsageStats) sendUsageStats(ctx context.Context) (string, error) {
 	if !uss.Cfg.ReportingEnabled {
-		return nil
+		return "", nil
 	}
-
-	uss.log.Debug("Sending anonymous usage stats", "url", usageStatsURL)
+	ctx, span := uss.tracer.Start(ctx, "UsageStats.BackgroundJob")
+	defer span.End()
+	traceID := tracing.TraceIDFromContext(ctx, false)
+	uss.log.FromContext(ctx).Debug("Sending anonymous usage stats", "url", usageStatsURL)
+	start := time.Now()
 
 	report, err := uss.GetUsageReport(ctx)
 	if err != nil {
-		return err
+		return traceID, err
 	}
 
 	out, err := json.MarshalIndent(report, "", " ")
 	if err != nil {
-		return err
+		return traceID, err
 	}
 
 	data := bytes.NewBuffer(out)
-	sendUsageStats(uss, data)
-	return nil
+	err = sendUsageStats(uss, ctx, data)
+	if err != nil {
+		return traceID, err
+	}
+
+	uss.log.FromContext(ctx).Info("Sent usage stats", "duration", time.Since(start))
+	return traceID, nil
 }
 
 // sendUsageStats sends usage statistics.
 //
 // Stubbable by tests.
-var sendUsageStats = func(uss *UsageStats, data *bytes.Buffer) {
-	go func() {
-		client := http.Client{Timeout: 5 * time.Second}
-		resp, err := client.Post(usageStatsURL, "application/json", data)
-		if err != nil {
-			uss.log.Error("Failed to send usage stats", "err", err)
-			return
-		}
-		if err := resp.Body.Close(); err != nil {
-			uss.log.Warn("Failed to close response body", "err", err)
-		}
-	}()
+var sendUsageStats = func(uss *UsageStats, ctx context.Context, data *bytes.Buffer) error {
+	ctx, span := uss.tracer.Start(ctx, "UsageStats.Send")
+	defer span.End()
+
+	client := http.Client{Timeout: 5 * time.Second}
+	resp, err := client.Post(usageStatsURL, "application/json", data)
```

Comment:
Maybe should include the `ctx` for the outgoing request? Should we inject trace headers for the outgoing request?

Reply:
Are you able to link to an example of where that's been done?

Reply:
```suggestion
	req, err := http.NewRequestWithContext(ctx, http.MethodPost, usageStatsURL, data)
	req.Header.Set("Content-Type", "application/json")
	uss.tracer.Inject(ctx, req.Header, span)
	resp, err := client.Do(req)
```

Like this?

Reply:
https://github.com/grafana/grafana/blob/e714d750f2fc31ee50db4d347be8900bbef58b2a/pkg/api/pluginproxy/ds_proxy.go#L149

Reply:
https://github.com/grafana/grafana/blob/e714d750f2fc31ee50db4d347be8900bbef58b2a/pkg/tsdb/graphite/graphite.go#L155

Reply:
Yes, that looks correct :+1:

Reply:
Yes, could be risky to introduce since it's only used for datasources currently. Would suggest considering that in a separate PR together with tests

Reply:
I kind of thought so, I like the idea of using it, but agree that we'll wait til a separate PR